src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -424,5 +425,51 @@ trait SubmissionComponent {
         where w.submission_id = ${submissionId}""".as[WorkflowInputResolutionListResult]
       }
     }
+
+    object DeleteSubmissionQuery extends RawSqlQuery {
+      val driver: JdbcDriver = SubmissionComponent.this.driver
+
+      def deleteAction(workspaceId: UUID) = {
+
+        def workspaceSubmissionQuery(workflow: SQLActionBuilder, join: SQLActionBuilder) = {
+          concatSqlActions(sql" inner join ", workflow, sql" w on ", join,
+            sql" inner join SUBMISSION s on w.submission_id=s.id where s.workspace_id=${workspaceId}")
+        }
+
+        val tables: Seq[SQLActionBuilder] = Seq(sql"WORKFLOW", sql"WORKFLOW_FAILURE")
+
+        DBIO.seq(concatSqlActions(
+          // delete submission attributes, workflow messages, and submission validations for workflows and
+          // workflow failures
+          sql"""delete sa from SUBMISSION_ATTRIBUTE sa
+                 inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
+               """, workspaceSubmissionQuery(sql"WORKFLOW", sql"sv.workflow_id=w.id")).as[Int],
+          concatSqlActions(
+            sql"""delete sa from SUBMISSION_ATTRIBUTE sa
+                 inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
+               """, workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"sv.workflow_failure_id=w.id")).as[Int],
+          concatSqlActions(sql"delete wm from WORKFLOW_MESSAGE wm ",
+            workspaceSubmissionQuery(sql"WORKFLOW", sql"wm.workflow_id=w.id")).as[Int],
+          concatSqlActions(sql"delete wm from WORKFLOW_MESSAGE wm ",
+            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"wm.workflow_id=w.id")).as[Int],
+          concatSqlActions(sql"delete sv from SUBMISSION_VALIDATION sv",
+            workspaceSubmissionQuery(sql"WORKFLOW", sql"sv.workflow_id=w.id")).as[Int],
+          concatSqlActions(sql"delete sv from SUBMISSION_VALIDATION sv", // JOIN IS DIFFERENT FOR WORKFLOW_FAILURE_ID
+            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"sv.workflow_failure_id=w.id")).as[Int],
+          concatSqlActions(sql"delete we from WORKFLOW_ERROR we",
+            workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"w.id=we.workflow_failure_id")).as[Int]
+        ) andThen {
+          DBIO.sequence(tables map { workflow_table =>
+            // delete workflows and workflow failure
+            concatSqlActions(sql"delete w from ", workflow_table,
+              sql""" w inner join SUBMISSION s on w.submission_id=s.id
+              where s.workspace_id=${workspaceId}""").as[Int]
+          })
+        } andThen {
+          // delete submissions
+          sqlu"delete s from SUBMISSION s where s.workspace_id=${workspaceId}"
```

Comment:
This doesn't need to be raw SQL.  Remove this and update `WorkspaceComponent.deleteWorkspaceSubmissions` to match `WorkspaceComponent.deleteWorkspaceEntitiesAndAttributes`